File: gl-frontend/src/models/domain/BannerGacha.ts
import { NikkeItem } from './Nikke'

export interface BannerGacha {
  id?: number
  date: string
  pickUpName: string
  gemsUsed: number
  ticketUsed: number
  totalAttempt: number
  totalSSR: number
  nikkePulled: NikkeItem[]
}

File: gl-rapi/src/main/kotlin/me/gachalyfe/rapi/domain/model/BannerGacha.kt
package me.gachalyfe.rapi.domain.model

data class BannerGacha(
    val id: Long,
    val date: String,
    val pickUpName: String,
    val gemsUsed: Int,
    val ticketUsed: Int,
    val totalAttempt: Int,
    val totalSSR: Int,
    val nikkePulled: List<Nikke> = emptyList(),
)
